src/main/java/baseball/domain/Result.java

```diff
@@ -0,0 +1,30 @@
+package baseball.domain;
+
+import baseball.presentation.Message;
+
+public class Result {
+    Message message = new Message();
+
+
+    void printResult(int[] result) {
+        int strike = result[0];
+        int ball = result[1];
+
+        if (strike == 3) {
+            message.printStrikeCaseMessage(strike);
+        } else {
+            if (ball == 0 && strike == 0) {
+                message.nothingCaseMessage();
+            }
+            if (ball > 0 && strike > 0) {
+                message.printballStrikeCaseMessage(ball, strike);
+            }
+            if (ball > 0 && strike == 0) {
+                message.printBallCaseMessage(ball);
+            }
+            if (strike > 0 && ball == 0) {
+                message.printStrikeCaseMessage(strike);
+            }
+        }
+    }
```

Comment:
해당 로직은 message 내에 들어 있는 게 좋을 것 같습니다. 팩토리 패턴이 잘못 사용되고 있어요.
내부에서 충분히 처리 가능한 로직이, 팩토리 패턴을 위해서 외부로 노출되어 있네요.
해당 로직을 message 내로 보내면 아래와 같이 깔끔한 코드가 완성됩니다.
```suggestion
    void printResult(int[] result) {
        int strike = result[0];
        int ball = result[1];
        System.out.println(new Message(strike, ball));
    }
```